src/api/Controllers/UploadController.cs

```diff
@@ -147,6 +153,54 @@ public async Task<ActionResult<int>> UploadFileAsync(int workgroupId, IFormFile
                 }
             }
 
+            // Add lithology imports if provided
+            if (lithologyImports != null)
+            {
+                // Get the kind id of a lithostratigraphy.
+                var lithoStratiKindId = context.Codelists.FirstOrDefault(cl => cl.Schema == "layer_kind" && cl.IsDefault == true).Id;
+
+                // Group lithology records by import id to get lithologies by boreholes.
+                var boreholeGroups = lithologyImports.GroupBy(l => l.ImportId);
+
+                var stratiesToAdd = new List<Stratigraphy>();
+                var lithologiesToAdd = new List<Layer>();
+                foreach (var boreholeLithologies in boreholeGroups)
+                {
+                    // Group lithology of one borehole by strati import id to get lithologies per stratigraphy.
+                    var stratiGroups = boreholeLithologies.GroupBy(bhoGroup => bhoGroup.StratiImportId);
+                    foreach (var stratiGroup in stratiGroups)
+                    {
+                        // Create a stratigraphy and assign it to the borehole with the provided import id.
+                        var strati = new Stratigraphy()
```

Comment:
PP: Mit dem Initialisierungssyntax kann man die `()` weglassen:
```suggestion
                        var strati = new Stratigraphy
```